src/main/java/org/jvnet/hudson/update_center/IndexHtmlBuilder.java

```diff
@@ -69,16 +72,41 @@ public IndexHtmlBuilder(PrintWriter out, String title) {
         );
     }
 
+    private String base64ToHex(String base64) {
+        byte[] decodedBase64 = Base64.decode(base64.getBytes());
+        return Hex.encodeHexString(decodedBase64);
+    }
+
     public void add(MavenArtifact a) throws IOException {
-        add(a.getURL().getPath(), a.version);
+        ArtifactSource.Digests digests = a.getDigests();
+        if (digests == null) {
+            return;
+        }
+        String checksums = "SHA-1: " + base64ToHex(digests.sha1);
+        if (digests.sha256 != null) {
+            checksums += ", SHA-256: " + base64ToHex(digests.sha256);
+        }
+        add(a.getURL().getPath(), a.version, checksums);
     }
 
     public void add(String url, String caption) throws MalformedURLException {
         out.println(
-            "<tr><td><img src='http://jenkins-ci.org/images/jar.png'/></td><td><a href='"+ url +"'>"+ caption +"</a></td></tr>"
+                "<tr><td><img src='http://jenkins-ci.org/images/jar.png'/></td><td><a href='"+ url +"'>"+ caption +"</a></td></tr>"
         );
     }
 
+    public void add(String url, String caption, String metadata) throws MalformedURLException {
+        if (metadata == null) {
+            out.println(
```

Comment:
Could just call `add(url, caption)` instead of duplicating 😄

Reply:
I think I meant to make `add(String, String)` call this with a `null` third arg. Unsure. This went through a few iterations as I wasn't sure whether the third arg should bring its `<td>` wrapper or not.